mantle/kola/harness.go

```diff
@@ -924,10 +977,104 @@ func RegisterExternalTests(dir string) error {
 	return RegisterExternalTestsWithPrefix(dir, basename)
 }
 
+// Create a parent test that runs non-exclusive tests as subtests
+func makeNonExclusiveTest(tests []*register.Test, config *conf.UserData, flight platform.Flight) (register.Test, func(h *harness.H)) {
+	// Parse test flags
+	var (
+		internetAccess     bool
+		noSSHKeyInMetadata bool
+		noSSHKeyInUserdata bool
+	)
+	var flags []register.Flag
+	for _, test := range tests {
+		if !noSSHKeyInMetadata && test.HasFlag(register.NoSSHKeyInMetadata) {
+			flags = append(flags, register.NoSSHKeyInMetadata)
+			noSSHKeyInMetadata = true
+		}
+		if !noSSHKeyInUserdata && test.HasFlag(register.NoSSHKeyInUserData) {
+			flags = append(flags, register.NoSSHKeyInUserData)
+			noSSHKeyInUserdata = true
+		}
+		if !internetAccess && testRequiresInternet(test) {
+			flags = append(flags, register.RequiresInternetAccess)
+			internetAccess = true
+		}
+	}
+
+	nonExclusiveWrapper := register.Test{
+		Name: "non-exclusive-tests",
+		Run: func(c cluster.TestCluster) {
+			// noop
+			_ = c
+		},
+		UserData: config,
+		// This will allow runTest to copy kolet to machine
+		NativeFuncs: make(map[string]register.NativeFuncWrap),
+		ClusterSize: 1,
+		Flags:       flags,
+	}
+
+	// Each non-exclusive test is run as a subtest of this wrapper test
+	runNonExclusive := func(parentH *harness.H) {
+		c, destroyCluster := runTest(parentH, &nonExclusiveWrapper, "", flight)
```

Comment:
If we rely on the `runTest` closure that already exists, then our `Run` function will already have available the VM provisioned for us by the harness, and we shouldn't need to change when the cluster gets destroyed.

Reply:
Hmm yes you're right. I initially thought that the parent harness was not accessible from the `Run` function but I see that I access it via `TestCluster.H`. I'll change this.